Polyhedron/demo/Polyhedron/Plugins/PCA/Clipping_box_plugin.cpp

```diff
@@ -148,6 +159,25 @@ void Clipping_box_plugin::enableAction() {
 }
 
 void Clipping_box_plugin::clip(bool b)
+{
+  clipping = b;
+  if (b)
+  {
+    Q_FOREACH(CGAL::QGLViewer* v, CGAL::QGLViewer::QGLViewerPool())
+    {
+      v->installEventFilter(this);
+    }
+  }
+  else {
+    Q_FOREACH(CGAL::QGLViewer* v, CGAL::QGLViewer::QGLViewerPool())
+    {
+      v->removeEventFilter(this);
+    }
```

Comment:
There are a lot of uses of `Q_FOREACH` in our code base, but new code must not use it. Use C++11 for-loop instead:

```suggestion
    for(CGAL::QGLViewer* v: CGAL::QGLViewer::QGLViewerPool())
    {
      v->installEventFilter(this);
    }
  }
  else {
    for(CGAL::QGLViewer* v: CGAL::QGLViewer::QGLViewerPool())
    {
      v->removeEventFilter(this);
    }
```

In Qt6, there is a warning about `Q_FOREACH`, when the container the loop it iterating on is not a Qt container. And even Qt6's documentation advise to use C++ for-loop instead: https://doc.qt.io/qt-6/foreach-keyword.html